### PR TITLE
XW-3393 | Add support for blog posts

### PR DESCRIPTION
**Pull request by rogatty**
## Links

* https://wikia-inc.atlassian.net/browse/XW-3393
* https://github.com/Wikia/app/pull/12872

## Description

User blog posts will now be displayed in Mobile Wiki.

## Reviewers

@kvas-damian @hakubo @latata

### CODE REVIEW

File: app/controllers/blog.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+const {Controller, computed, inject} = Ember;
+
+export default Controller.extend({
+	application: inject.controller(),
+	article: inject.controller(),
+	wikiVariables: inject.service(),
+
+	commentsPage: computed.alias('application.commentsPage'),
+	mainPageTitle: computed.reads('wikiVariables.mainPageTitle'),
+	siteName: computed.reads('wikiVariables.siteName'),
```
Comment (kvas-damian) [inline app/controllers/blog.js]: I see a lot those attributes being copied from other controller. Maybe it's a time to create a mixin for them?

File: public/locales/en/main.json [hunk 2]
```diff
@@ -67,6 +67,9 @@
     "usages-header": "Appears on these pages",
     "usages-see-more-link-label": "See full list"
   },
+  "blog-page": {
+    "subtitle": "User blog"
```
Comment (kvas-damian) [inline public/locales/en/main.json:74]: Do we have a ticket for this trnaslation?

File: app/mixins/wiki-page-handler.js [hunk 3]
```diff
@@ -135,6 +136,15 @@ export default Ember.Mixin.create({
 			model.setData(data);
 
 			return model;
+		} else if (
+			currentNamespace === MediawikiNamespace.BLOG_ARTICLE &&
+			// User blog listing has BLOG_ARTICLE namespace but no article
```
Comment (kvas-damian) [inline app/mixins/wiki-page-handler.js:141]: So what happens when you visit such page?

Comment (rogatty) [reply app/mixins/wiki-page-handler.js:141]: You go to Oasis.

File: app/styles/module/_comments.scss [hunk 4]
```diff
@@ -34,6 +34,38 @@
 	@extend %list-reset;
 
 	font-size: rem-calc(13);
+
+	&__no-comments, &__add-comment {
+		display: flex;
```
Comment (latata) [inline app/styles/module/_comments.scss]: alphabetical order ;)

File: app/templates/components/article-comments.hbs [hunk 5]
```diff
@@ -1,17 +1,31 @@
...
-	</div>
-{{/if}}
+<div {{action 'toggleComments'}} class='show-comments-btn page-btn {{if isCollapsed 'collapsed'}}'>
+	{{i18n 'article.comments-label' count=commentsCount}}
+	{{svg 'chevron' viewBox='0 0 12 7' class='icon chevron'}}
+</div>
 {{#if showComments}}
 	{{#if model.comments}}
-		<button class='page-btn {{unless prevButtonShown 'hidden'}}' {{action 'prevPage'}}>{{i18n 'article-comments.prev-button'}}</button>
+		<button class='page-btn {{unless prevButtonShown 'hidden'}}' {{action 'prevPage'}}>
```
Comment (latata) [inline app/templates/components/article-comments.hbs]: `"` instead of `'` for attribute values

**Review by rybmat — APPROVED** [review]

File: app/styles/module/_comments.scss [hunk 6]
```diff
@@ -34,6 +34,38 @@
...
+		color: $wds-color-slate-gray;
+		font-size: $wds-typescale-size-minus-1;
+		padding: 20px 0;
+	}
+
+	&__add-comment {
+		padding: 10px 0 20px;
+	}
+
+	&__replay-icon {
```
Comment (rogatty) [inline app/styles/module/_comments.scss]: replay or reply?

**Pull request merged**